File: CsLox/com/craftinginterpreters/lox/Clock.cs
using com.craftinginterpreters.lox;
using System;
using System.Collections.Generic;
using System.Data.SqlTypes;
using System.Text;

namespace com.craftinginterpreters.lox
{
    /// <summary>
    /// 
    /// </summary>
    class Clock : LoxCallable
    {
        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public int arity() { 
            return 0; 
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="interpreter"></param>
        /// <param name="arguments"></param>
        /// <returns></returns>
        public Object call(Interpreter interpreter, List<Object> arguments)
        {
            if (Settings.TIMING == Settings.TimingTypes.TICKS_FROM_APP_START)
            {
                return (double)System.Environment.TickCount;
            }
            else if (Settings.TIMING == Settings.TimingTypes.SECONDS_FROM_EPOCH)
            {
                return (double)DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
            }
            else
            {
                return (double)DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() / 1000.0;
            }
        }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public override string ToString()
        {
            return "<native fn>";
        }
    }
}

File: JavaLox/src/com/craftinginterpreters/lox/Clock.java
package com.craftinginterpreters.lox;

import java.util.List;

/**
 *
 * @author brusc
 */
public class Clock implements LoxCallable {

   /**
    *
    * @return
    */
   @Override
   public int arity() {
      return 0;
   }

   /**
    *
    * @param interpreter
    * @param arguments
    * @return
    */
   @Override
   public Object call(Interpreter interpreter, List<Object> arguments) {      
      if (Settings.TIMING == Settings.TimingTypes.TICKS_FROM_APP_START) {
         return (double) System.nanoTime();
      } else if (Settings.TIMING == Settings.TimingTypes.SECONDS_FROM_EPOCH) {
         return (double) System.currentTimeMillis();
      } else {
         return (double) System.currentTimeMillis() / 1000.0;
      }
   }

   /**
    *
    * @return
    */
   @Override
   public String toString() {
      return "<native fn>";
   }
}
